bayesian_cnn_prometheus/analysis/correlation_analyzer.py

```diff
@@ -0,0 +1,159 @@
+        :param save_to_json: if True, save metrics to JSON file
+        :param print_metrics: if True, prints all of the numeric metrics
+        :param plot_metrics: if True, plots all of the displayable metrics
+        """
+        self._assign_adapted_rand_error_metrics()
+        self._assign_variation_of_information_metrics()
+        self._assign_hausdorff_distance()
+        self._assign_mean_squared_error()
+        self._assign_normalized_root_mse()
+        self._assign_peak_signal_noise_ratio()
+        # self._assign_structural_similarity_metrics()
+        # self._assign_contingency_table()
+        # self._assign_cross_correlate()
+
+        if save_to_json:
+            self._save_to_json()
+
+        if print_metrics:
+            self._print_metrics()
+
+        if plot_metrics:
+            self._plot_metrics()
+
+    def _save_to_json(self):
+        """
+        Saves metrics to a JSON file
+        """
+        if self.metrics:
+            with open('metrics.json', 'w') as f:
+                json.dump(self.metrics, f)
+
+    def _print_metrics(self):
+        """
+        Prints all of the numeric metrics.
+        """
+        if self.metrics:
+            print("Metrics:")
+            print(f"- Adapted Rand error: {self.metrics['are']}")
+            print(f"- precision: {self.metrics['precision']}")
+            print(f"- recall: {self.metrics['recall']}")
+            print(f"- cancer|variance masks conditional entropy: {self.metrics['cancer_to_variance_conditional_entropy']}")
+            print(f"- variance|cancer masks conditional entropy: {self.metrics['variance_to_cancer_conditional_entropy']}")
+            print(f"- the Hausdorff distance: {self.metrics['hausdorff_distance']}")
+            print(f"- Mean Squared Error: {self.metrics['mse']}")
+            print(f"- Normalized Root Mean Squared Error: {self.metrics['normalized_root_mse']}")
+            print(f"- peak signal to noise ratio: {self.metrics['peak_signal_noise_ratio']}")
+            # print(f"- mean structural similarity index: {self.metrics['ss_mean']}")
+
+    def _plot_metrics(self):
+        """
+        Plots all of the displayable metrics.
+        """
+        if self.metrics:
+            fig, axis = plt.subplots(1, 4)
+            axis[0].plot(self.metrics["contingency_table"])
+            axis[0].set_title("Contingency table")
+            axis[1].plot(self.metrics["ss_image"])
+            axis[1].set_title("Structural similarity image")
+            axis[2].plot(self.metrics["ss_gradient"])
+            axis[2].set_title("Structural similarity gradient")
+            axis[3].plot(self.metrics["cross_correlate"])
+            axis[3].set_title("Cross-correlate array")
+
+    def _assign_adapted_rand_error_metrics(self):
+        """
+        Computes Adapted Rand error, precision and recall.
+        """
+        self.metrics["are"], self.metrics["precision"], self.metrics["recall"] = \
```

Comment:
Jakby Ci się chciało to ładnie byłoby ujednolicić podwójne i pojedyncze ' " bo są różnie :)

Reply:
Myślę, że zamiast tego lepiej by było dodać linter. 
Np. w `GithubActions` albo jak hook go gita. 
Główna motywacja jest taka, że wtedy nie my będziemy musieli się o to martwić. 
Oprócz tego w PR nie będą pojawiać się zmiany w formatowaniu (jak np. skrócenie linii)

Reply:
Pojedyncze cudzysłowy były mi potrzebne tylko przy wyświetlaniu metryk. Po sugestii @Qwebeck a propos nie ma już tego problemu :)